### PR TITLE
Grafana/toolkit: Fix incorrect image and font generation for plugin builds

### LINKED ISSUE #51953: Custom fonts for plugins doesn't work after v9
**What happened**:
When using the `@grafana/toolkit` `9.0.2` custom fonts don't load.
image

**What you expected to happen**:
Custom fonts to load.
image

**How to reproduce it (as minimally and precisely as possible)**:
Repo: https://github.com/ZuperZee/zuperzee-font-panel

**Anything else we need to know?**:
Changing the toolkit to `8.5.6` loads the font properly.

**Environment**:
- Grafana version: v9.0.2
- Data source type & version:
- OS Grafana is installed on: Ubuntu 20.04 WSL (docker)
- User OS & Browser: Windows & Edge v103.0.1264.44 
- Grafana plugins:
- Others:

### PR DESCRIPTION
**Pull request by academo**
**What this PR does / why we need it**:
Replace webpack file-loader with asset/resource

The existing configuration is broken since the migration to webpack 5. It generates incorrect images and font files on the root folder with references to non-existent paths (see https://github.com/grafana/grafana/issues/51953)

This fix uses the new recommended `asset` module from webpack 5 and removes the existing logic to inline images inside the css using base64 encoding.

**Which issue(s) this PR fixes**:

Closes: https://github.com/grafana/grafana/issues/51953

### CODE REVIEW

File: packages/grafana-toolkit/src/config/webpack/loaders.ts [hunk 1]
```diff
@@ -33,39 +31,6 @@ export const getStylesheetEntries = (root: string = process.cwd()) => {
   return entries;
 };
 
-export const hasThemeStylesheets = (root: string = process.cwd()) => {
```
Comment (academo) [inline packages/grafana-toolkit/src/config/webpack/loaders.ts:36]: This function is not longer used, originally it was trying to figure if the developer forgot to add a light/dark theme in the plugin but this is "too smart" for the builder to do.

File: packages/grafana-toolkit/src/config/webpack/loaders.ts [hunk 2]
```diff
@@ -139,34 +104,21 @@ export const getStyleLoaders = () => {
...
       test: /\.(woff|woff2|eot|ttf|otf)(\?v=\d+\.\d+\.\d+)?$/,
-      loader: require.resolve('file-loader'),
-      options: {
-        // Keep publicPath relative for host.com/grafana/ deployments
-        publicPath: `public/plugins/${getPluginId()}/fonts`,
-        outputPath: 'fonts',
-        name: '[name].[ext]',
+      type: 'asset/resource',
+      generator: {
+        publicPath: `fonts/`,
```
Comment (tolzhabayev) [inline packages/grafana-toolkit/src/config/webpack/loaders.ts:120]: I am not sure why we had `public/plugins/.../fonts` here before and if there is anything we need to consider here @jackw

**Review by tolzhabayev — APPROVED** [review]
From my POV it looks good, just one things to clarify about the `publicPath`

File: packages/grafana-toolkit/src/config/webpack/loaders.ts (hunk 2 above)
Comment (academo) [reply packages/grafana-toolkit/src/config/webpack/loaders.ts:120]: @jackw and I looked at this together and using just this path generates the correct file output and refs.

Comment (jackw) [reply packages/grafana-toolkit/src/config/webpack/loaders.ts:120]: @tolzhabayev I don't know for sure but it seems like; there were some changes to the way these loaders worked out paths between `webpack@4` and `webpack@5`. Whilst testing with @academo it looks like these paths resolve themselves relatively which we believe will give the most consistent path resolution.

**Review by jackw — APPROVED** [review]
Awesome work on this! 🚀

**Pull request merged**